business/core/home/stores/homedb/homedb.go

```diff
@@ -0,0 +1,216 @@
+package homedb
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ardanlabs/service/business/core/home"
+	db "github.com/ardanlabs/service/business/data/dbsql/pq"
+	"github.com/ardanlabs/service/business/data/order"
+	"github.com/ardanlabs/service/business/data/transaction"
+	"github.com/ardanlabs/service/foundation/logger"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+)
+
+// Store manages the set of APIs for home database access.
+type Store struct {
+	log *logger.Logger
+	db  sqlx.ExtContext
+}
+
+// NewStore constructs the api for data access.
+func NewStore(log *logger.Logger, db *sqlx.DB) *Store {
+	return &Store{
+		log: log,
+		db:  db,
+	}
+}
+
+// ExecuteUnderTransaction constructs a new Store value replacing the sqlx DB
+// value with a sqlx DB value that is currently inside a transaction.
+func (s *Store) ExecuteUnderTransaction(tx transaction.Transaction) (home.Storer, error) {
+	ec, err := db.GetExtContext(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	s = &Store{
+		log: s.log,
+		db:  ec,
+	}
+
+	return s, nil
+}
+
+// Create adds a Home to the database. It returns an error if something went wrong
+func (s *Store) Create(ctx context.Context, hme home.Home) error {
+	const q = `
+      INSERT INTO homes
+        (home_id, user_id, type, address_1, address_2, zip_code, city, state, country, da:te_created, date_updated)
+      VALUES
+      (:home_id, :user_id, :type, :address_1, :address_2, :zip_code, :city, :state, :country, :date_created, :date_updated)
+    `
+
+	if err := db.NamedExecContext(ctx, s.log, s.db, q, toDBHome(hme)); err != nil {
+		return fmt.Errorf("namedexeccontext: %w", err)
+	}
+
+	return nil
+}
+
+// Delete removes the Home identified by a given ID.
+func (s *Store) Delete(ctx context.Context, hme home.Home) error {
+	data := struct {
+		ID string `db:"product_id"`
+	}{
+		ID: hme.ID.String(),
+	}
+
+	const q = `
+	DELETE FROM
+		products
+	WHERE
+		product_id = :product_id`
+
+	if err := db.NamedExecContext(ctx, s.log, s.db, q, data); err != nil {
+		return fmt.Errorf("namedexeccontext: %w", err)
+	}
+
+	return nil
+}
+
+// Query retrieves a list of existing homes from the database.
+func (s *Store) Query(ctx context.Context, filter home.QueryFilter, orderBy order.By, pageNumber int, rowsPerPage int) ([]home.Home, error) {
+	data := map[string]interface{}{
+		"offset":        (pageNumber - 1) * rowsPerPage,
+		"rows_per_page": rowsPerPage,
+	}
+
+	const q = `
+	SELECT
+		*
```

Comment:
This * burned us a few times on TP. We should fix this everywhere and list out the fields.

Reply:
Will do that for all domains then.